src/msg/simple/Pipe.cc

```diff
@@ -1115,7 +1115,7 @@ int Pipe::connect()
   port = peer_addr_for_me.get_port();
 
   ldout(msgr->cct,20) << "connect read peer addr " << paddr << " on socket " << sd << dendl;
-  if (peer_addr != paddr) {
+  if (!peer_addr.is_same_host(paddr)) {
```

Comment:
is this change needed with the other adjustments?

Reply:
@liewegas 
I was actually going to rewrite all the compare tests, to exclude any dependancies on `{sa,ss}_len` but that turns out to be relatively messy. :( So this was the next best thing.
I'll rerun the tests with `!=` there, and amend if it runs oke with `cep_test_msgr`